stock_available_to_promise_release/models/stock_move.py

```diff
@@ -591,6 +597,39 @@ def _get_chained_moves_iterator(self, chain_field):
             yield moves
             moves = moves.mapped(chain_field)
 
+    def _return_quantity_in_stock(self, quantity_to_return):
+        picking = self.picking_id
+        picking.ensure_one()
+
+        returned_quantity = 0
+        # create return
+        return_type = picking.picking_type_id.return_picking_type_id
+        wiz_values = {
+            "picking_id": picking.id,
+            "original_location_id": picking.location_dest_id.id,
+            "location_id": return_type.default_location_dest_id.id,
+        }
+        product_return_moves = []
+        for move in self:
+            if not move.state == "done":
+                continue
+            move_qty = min(quantity_to_return - returned_quantity, move.quantity_done)
```

Comment:
Process real quantities and not uom quantities. You can use `product_qty` as the move is done
```suggestion
            move_qty = min(quantity_to_return - returned_quantity, move.product_qty)
```